src/plugins/mer/merdeploysteps.cpp

```diff
@@ -733,6 +745,23 @@ bool MerMb2RpmBuildStep::init()
             QLatin1String(Sfdk::Constants::WRAPPER_RPM));
     CommandLine rpmCommand(rpm);
     rpmCommand.addArgs(pp->command().arguments(), CommandLine::Raw);
+
+    if (m_signPackages) {
+        rpmCommand.addArgs("--sign", CommandLine::Raw);
+
+        QStringList packageUserSigningArgs;
+        if (!m_signPackageKey.isEmpty())
+            packageUserSigningArgs.append(QStringLiteral("-c") + QtcProcess::quoteArgUnix("package-signing-user=%1").arg(m_signPackageKey));
+        if (!m_signPackagePassFile.isEmpty())
+            packageUserSigningArgs.append(QStringLiteral("-c") + QtcProcess::quoteArgUnix("package-signing-passphrase-file=%1").arg(m_signPackagePassFile));
+
+        if (!packageUserSigningArgs.empty()) {
+            auto env = pp->environment();
+            env.appendOrSet(QLatin1String(Sfdk::Constants::MER_SSH_SFDK_OPTIONS), packageUserSigningArgs.join(' '));
+            pp->setEnvironment(env);
```

Comment:
This looks really hacky so I checked a bit deeper and I realized all the signing related configuration should be implemented in MerBuildConfigurationAspect/Widget instead. What should remain here is just something like:

```
    auto aspect = buildConfiguuration()->aspect<MerBuildConfigurationAspect>();
    QTC_ASSERT(aspect, return false);
    if (aspect->isSignPackagesEnabled())
        rpmCommand.addArgs("--sign", CommandLine::Raw);
```